### ISSUE TITLE
Does phpcbf fix comments?

### ISSUE THREAD

**Issue by bluestarstudios** [WordPress/WordPress-Coding-Standards#1759]
I have latest WordPress Coding Standards setup in PHPStorm. PHPCS reports that `You must use "/**" style comments for a file comment.` however when running phpcbf it doesn't seem to be able to fix that.  Same goes for warnings like `Inline comments must end in full-stops, exclamation marks, or question marks`. Is that normal?

Thanks!

**Comment by GaryJones** [comment]
> PHPCS reports that `You must use "/**" style comments for a file comment.` however when running phpcbf it doesn't seem to be able to fix that. 

You can see [in the `WordPress-Docs` ruleset](https://github.com/WordPress/WordPress-Coding-Standards/blob/develop/WordPress-Docs/ruleset.xml#L11) that the file comments checks come from the Squiz standard, which is built into PHPCS.

The violation message you see can be found [here](https://github.com/squizlabs/PHP_CodeSniffer/blob/f78fa6f75c47d0ec289114ed0467f8e98774ee28/src/Standards/Squiz/Sniffs/Commenting/FileCommentSniff.php#L56), and whilst it gives the error (`addError()`), and records the metric (`recordMetric()`), there's no auto-fix (you'd be looking to an `addFix()`).

I would guess that in this case, an auto-fix wouldn't make sense - take this example:

```php
<?php

// Pretend file comment.

/**
 * Foo.
 */
function foo() {
	return 'foo';
}

// I am an inline comment

foo();
```

Here, changing that first comment to have the `/** ... */` format would simply throw a whole load more violations about what is expected of file DocBlocks (presence of certain tags, or formatting, etc.). 

> Same goes for warnings like `Inline comments must end in full-stops, exclamation marks, or question marks`.

The reason for the lack of auto-fix here is a little simpler - how would the auto-fixer know whether to add a full-stop, exclamation or question mark?

To see easier whether an auto fix is available, look at the output:

<img width="951" alt="Screenshot 2019-07-06 at 15 17 08" src="https://user-images.githubusercontent.com/88371/60757383-bb578f80-a001-11e9-97ff-8fb693428c1d.png">

See that one of the items has an `[x]`? That means that `phpcbf` will auto-fix it.

**Comment by bluestarstudios** [comment]
Hi @GaryJones,

Thanks for the quick reply and the awesome explanation. That makes sense.
In my case, the particular code that generates the  `You must use "/**" style comments for a file comment` is as follows:

        <?php
        
        /*
        Plugin Name: Appointments+ Overrides
        Description: This plugin overrides the necessary functions to make the Appointments+ plugin function the way we need it for our website.
        Version: 1.3
        Author: David
        */
        
        if ( ! defined( 'ABSPATH' ) ) {
        	exit();
        }

I would assume this would be something that could be auto-fixed, no? (I'm using phpcs inside of PHPStorm, so I don't know how to view the output to see if there is an auto-fix for it).

**Comment by GaryJones** [comment]
> I would assume this would be something that could be auto-fixed, no?

In terms of changing `/*` to `/**`, it could, but as stated before, this would then trigger other errors - [no blank line before the file comment](https://github.com/squizlabs/PHP_CodeSniffer/blob/f78fa6f75c47d0ec289114ed0467f8e98774ee28/src/Standards/Squiz/Sniffs/Commenting/FileCommentSniff.php#L112), for example. Although WPCS currently [ignores](https://github.com/WordPress/WordPress-Coding-Standards/blob/d00b44ba67ae47faf7d2727faf0c9485a1e7f1d9/WordPress-Docs/ruleset.xml#L34-L51) a bunch of the Squiz checks, that may not always be the case. Any suggestion of auto-fix you want to make should be done by opening an issue upstream in the PHPCS repo.

If you want a file comment that includes the plugin headers, then look at the second example of https://codex.wordpress.org/File_Header#Plugin_File_Header_Example:

```php
<?php
/**
 * Plugin Name
 *
 * @package     PluginPackage
 * @author      Your Name
 * @copyright   2019 Your Name or Company Name
 * @license     GPL-2.0-or-later
 *
 * @wordpress-plugin
 * Plugin Name: Plugin Name
 * Plugin URI:  https://example.com/plugin-name
 * Description: Description of the plugin.
 * Version:     1.0.0
 * Author:      Your Name
 * Author URI:  https://example.com
 * Text Domain: plugin-slug
 * License:     GPL-2.0-or-later
 * License URI: http://www.gnu.org/licenses/gpl-2.0.txt
 */
```

If using PhpStorm, you can save that (or your version of it) as a template / code snippet, and then never have to worry about PHPCS for this again :-)

**Comment by bluestarstudios** [comment]
You're right, I totally opened the ticket in the wrong place! Thanks for all the clarifications! Have a great day!
